### PR TITLE
pango: Auto generate Language

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: pango/src/auto/language.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    pub fn to_str(&self) -> glib::GString {
+        unsafe {
+            from_glib_none(ffi::pango_language_to_string(mut_override(
+                self.to_glib_none().0,
+            )))
+        }
+    }
+
+    #[doc(alias = "pango_language_from_string")]
+    pub fn from_string(language: Option<&str>) -> Option<Language> {
```
Comment (bilelmoussaoui) [inline pango/src/auto/language.rs:62]: the function signature is more correct now, maybe we should use Result<T> as everywhere else and impl FromStr?

File: pango/src/auto/language.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            from_glib(ffi::pango_language_matches(
+                mut_override(self.to_glib_none().0),
+                range_list.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "pango_language_to_string")]
+    #[doc(alias = "to_string")]
+    pub fn to_str(&self) -> glib::GString {
```
Comment (bilelmoussaoui) [inline pango/src/auto/language.rs:53]: The function has been renamed from to_string to to_str

File: pango/src/auto/language.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    #[allow(clippy::should_implement_trait)]
+    pub fn default() -> Language {
+        unsafe { from_glib_none(ffi::pango_language_get_default()) }
+    }
+
+    #[cfg(any(feature = "v1_48", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v1_48")))]
+    #[doc(alias = "pango_language_get_preferred")]
+    #[doc(alias = "get_preferred")]
+    pub fn preferred() -> Option<Language> {
```
Comment (bilelmoussaoui) [inline pango/src/auto/language.rs:77]: oups this function is wrong and has to stay manual

File: pango/src/auto/language.rs (hunk 1 above)
Comment (sdroege) [reply pango/src/auto/language.rs]: Yes that would seem useful

File: pango/src/auto/language.rs (hunk 2 above)
Comment (sdroege) [reply pango/src/auto/language.rs:53]: Sounds better

File: pango/Gir.toml [hunk 4]
```diff
@@ -332,6 +331,22 @@ concurrency = "send+sync"
         name = "glyphs"
         const = true
 
+[[object]]
+name = "Pango.Language"
+status = "generate"
+    [[object.function]]
+    pattern = "get_(scripts|preferred)"
+    manual = true
```
Comment (sdroege) [inline pango/Gir.toml:339]: Add a comment here why. I think this needs an annotation fix, `(array zero-terminated=1)`

**Review by sdroege — COMMENTED** [review]
Seems good otherwise. Didn't you want to do the same for font description?

File: pango/Gir.toml (hunk 4 above)
Comment (bilelmoussaoui) [reply pango/Gir.toml]: right, done

**Review by sdroege — APPROVED** [review]

**Comment by bilelmoussaoui** [comment]
gnome gitlab is down, we can safely land this

File: pango/Gir.toml [hunk 5]
```diff
@@ -332,6 +331,22 @@ concurrency = "send+sync"
         name = "glyphs"
         const = true
 
+[[object]]
+name = "Pango.Language"
+status = "generate"
+    [[object.function]]
+    pattern = "get_(scripts|preferred)"
+    manual = true # probably needs annotation fixes upstream (array zero-terminated=1)
```
Comment (sdroege) [inline pango/Gir.toml:339]: https://gitlab.gnome.org/GNOME/pango/-/merge_requests/655

**Pull request merged**